### PR TITLE
add filter_sample_categories and allow slice in __getitem__

### PR DESCRIPTION
**Pull request by RNAer**
delete extra data/notebooks plus other minor changes

### CODE REVIEW

**Comment by amnona** [comment]
unit testing is failing (one test has a problem in test_filtering...)

File: calour/analysis.py [hunk 1]
```diff
@@ -65,13 +65,14 @@ def correlation(exp, field, method='spearman', nonzero=False, transform=None, nu
         the desired FDR control level
     numperm : int
         number of permutations to perform
+    fdr_method : str
+        method to compute FDR. Allowed method include "", ""
```
Comment (amnona) [inline calour/analysis.py]: the FDR procedure to determine significant bacteria
        'dsfdr' : discrete FDR method
        'bhfdr' : Benjamini-Hochberg FDR method
        'byfdr' : Benjamini-Yekutielli FDR method
        'filterBH' : Benjamini-Hochberg FDR method with filtering (automatic removal of features which cannot reach significance level)

File: calour/filtering.py [hunk 2]
```diff
@@ -97,6 +98,44 @@ def downsample(exp, field, axis=0, num_keep=None, inplace=False):
...
+    Examples
+    --------
+
+    Parameters
+    ----------
+    field : str
+        The name of the column in samples metadata table. This column
+        should has categorical values
+    min_samples : int (optional)
+        Filter away the samples with a value in the given column if its sample count is
```
Comment (amnona) [inline calour/filtering.py]: Maybe default should be 0? so no dropping with min_samples unless specified?

**Comment by amnona** [comment]
added a few silly comments in the code, but looks good :)

**Comment by RNAer** [comment]
nope. if 0, then you don't need to run this function at all.

**Pull request merged**